fuel-p2p/src/service.rs

```diff
@@ -239,16 +243,28 @@ impl<Codec: NetworkCodec> FuelP2PService<Codec> {
         Ok(())
     }
 
-    pub async fn next_event(&mut self) -> FuelP2PEvent {
-        loop {
-            if let SwarmEvent::Behaviour(fuel_behaviour) =
-                self.swarm.select_next_some().await
-            {
-                if let Some(event) = self.handle_behaviour_event(fuel_behaviour) {
-                    return event
-                }
-            }
+    pub fn report_message_validation_result(
+        &mut self,
+        msg_id: &MessageId,
+        propagation_source: &PeerId,
+        acceptance: MessageAcceptance,
+    ) -> Result<bool, PublishError> {
+        self.swarm.behaviour_mut().report_message_validation_result(
+            msg_id,
+            propagation_source,
+            acceptance,
+        )
+    }
+
+    /// Handles P2P Events.
+    /// Returns only events that are of interest to the Network Orchestrator.
+    pub async fn next_event(&mut self) -> Option<FuelP2PEvent> {
+        if let SwarmEvent::Behaviour(fuel_behaviour) = self.swarm.select_next_some().await
+        {
+            return self.handle_behaviour_event(fuel_behaviour)
         }
+
+        None
     }
```

Comment:
[nit] unnessecary return

```suggestion
            self.handle_behaviour_event(fuel_behaviour)
        } else {
            None
        }
    }
```